src/services/ManualMapLoading.ts

```diff
@@ -0,0 +1,207 @@
+import fs from 'fs/promises'
+
+import { Logger } from '../Logger.js'
+import { Client } from '../client/Client.js'
+import config from '../../config/Config.js'
+import { MapService } from "./MapService.js"
+
+export class ManualMapLoading {
+  private static readonly prefix: string = config.mapsDirectoryPrefix
+  private static readonly stadium: boolean = config.stadiumOnly !== undefined ? config.stadiumOnly : process.env.SERVER_PACKMASK === "nations"
+  private static mapIndex: number = 0
+  private static oldQueue: tm.Map[]
+  private static oldCurr: tm.CurrentMap
+
+  static async getFileNames(): Promise<string[]> {
+    const files: string[] = await fs.readdir(this.prefix + config.mapsDirectory, { recursive: true })
+    return files.map(a => config.mapsDirectory + a)
+  }
+
+  /**
+   * Parse every map in the `config.mapsDirectoryPrefix/config.mapsDirectory
+   * @param presentMaps list of already existing maps to compare with
+   * @param dirname the directory name
+   * @returns list of maps in the same format that the server returns them
+   */
+  static async parseMaps(presentMaps: tm.Map[] = [], dirname: string = config.mapsDirectory) {
+    const filesOrDirs = await fs.readdir(this.prefix + dirname, {withFileTypes: true})
+    if (filesOrDirs.length > 5000) Logger.warn(`Trying to parse a large amount of maps (${filesOrDirs.length}), reading their info might take a while.`)
+    const remainingMaps: tm.Map[] = []
+    const addedMaps: tm.Map[] = []
+    for (const f of filesOrDirs) {
+      if (f.isDirectory()) {
+        const res = await this.parseMaps(presentMaps, dirname + f.name + '/')
+        res[0].forEach(a => remainingMaps.push(a))
+        res[1].forEach(a => addedMaps.push(a))
+        continue
+      }
+      const map = await this.parseMap(dirname + f.name, presentMaps, remainingMaps.concat(addedMaps))
+      if (map instanceof Error) {
+        if (map.message.startsWith("PARSEERROR")) Logger.warn(map.message)
+        continue
+      }
+      if ((map as tm.Map).id !== undefined) remainingMaps.push(map as tm.Map)
```

Comment:
please use the curly brackets for control statements i beg you this looks ugly and unreadable

Reply:
I disagree, if it's a one-liner there's no need for brackets

Reply:
the rest of the codebase uses {} even for oneliners, it just looks better imo and its actually visible where the blocks end